activestorage/app/models/active_storage/blob.rb

```diff
@@ -110,6 +110,8 @@ def text?
   # Hiding the +service_url+ behind a redirect also gives you the power to change services without updating all URLs. And
   # it allows permanent URLs that redirect to the +service_url+ to be cached in the view.
   def service_url(expires_in: service.url_expires_in, disposition: :inline, filename: self.filename, **options)
+    filename = ActiveStorage::Filename.wrap(filename || self.filename)
```

Comment:
`filename || self.filename` for nil case to use default value from `blob.filename`